### ISSUE TITLE
Login dialog for AUR voting does not accept + in the password but aur.archlinux.org does

### ISSUE THREAD

**Issue by LJNielsenDk** [aarnt/octopi#524]
Logging in in a browser works, but Octopi says the credentials are wrong.

Octopi 0.14.0-2 from Manjaro Community repo, Build date Tuesday, 8 November 2022 16.38.10 CET. Everything else is also fully updated.

Relevant output of `/usr/bin/octopi -d`:

```
AurVote::login(): Second post replied with:  "<!DOCTYPE html>\n<html xmlns=\"http://www.w3.org/1999/xhtml\"\n      xml:lang=\"en\" lang=\"en\" >\n    <head>\n    <meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\">\n\n    <!-- CSS -->\n    <link rel=\"stylesheet\" href=\"/static/css/archweb.css\">\n    <link rel=\"stylesheet\" href=\"/static/css/aurweb.css\">\n\n    <!-- Resources -->\n    <link rel=\"shortcut icon\" href=\"/static/images/favicon.ico\">\n\n    <!-- Alternate resources -->\n    <link rel=\"alternate\" type=\"application/rss+xml\"\n          title=\"Newest Packages RSS\" href=\"/rss\">\n\n    <!-- Include local typeahead -->\n    <script type=\"text/javascript\" src=\"/static/js/typeahead.js\"></script>\n\n    <title>AUR (en) - Login</title>\n</head>\n\n    <body>\n        <div id=\"archnavbar\" class=\"anb-aur\">\n    <div id=\"archnavbarlogo\">\n        <h1>\n            <a href=\"/\" title=\"Return to the main page\">Arch Linux User Repository</a>\n        </h1>\n    </div>\n    <div id=\"archnavbarmenu\">\n        <ul id=\"archnavbarlist\">\n            <li id=\"anb-home\"><a href=\"https://www.archlinux.org/\" title=\"Arch news, packages, projects and more\">Home</a></li>\n            <li id=\"anb-packages\"><a href=\"https://www.archlinux.org/packages/\" title=\"Arch Package Database\">Packages</a></li>\n            <li id=\"anb-forums\"><a href=\"https://bbs.archlinux.org/\" title=\"Community forums\">Forums</a></li>\n            <li id=\"anb-wiki\"><a href=\"https://wiki.archlinux.org/\" title=\"Community documentation\">Wiki</a></li>\n            <li id=\"anb-bugs\"><a href=\"https://bugs.archlinux.org/\" title=\"Report and track bugs\">Bugs</a></li>\n            <li id=\"anb-security\"><a href=\"https://security.archlinux.org/\" title=\"Arch Linux Security Tracker\">Security</a></li>\n            <li id=\"anb-aur\"><a href=\"/\" title=\"Arch Linux User Repository\">AUR</a></li>\n            <li id=\"anb-download\"><a href=\"https://www.archlinux.org/download/\" title=\"Get Arch Linux\">Download</a></li>\n        </ul>\n    </div>\n</div>\n        <div id=\"content\">\n    <div id=\"lang_sub\">\n    <form method=\"post\" action=\"/language\">\n        <fieldset>\n            <div>\n                <select id=\"id_setlang\" name=\"set_lang\">\n                    \n                    <option\n                        value=\"ar\"\n                        \n                    >\n                        العربية\n                    </option>\n                    \n                    <option\n                        value=\"ast\"\n                        \n                    >\n                        Asturianu\n                    </option>\n                    \n                    <option\n                        value=\"ca\"\n                        \n                    >\n                        Català\n                    </option>\n                    \n                    <option\n                        value=\"cs\"\n                        \n                    >\n                        Český\n                    </option>\n                    \n                    <option\n                        value=\"da\"\n                        \n                    >\n                        Dansk\n                    </option>\n                    \n                    <option\n                        value=\"de\"\n                        \n                    >\n                        Deutsch\n                    </option>\n                    \n                    <option\n                        value=\"el\"\n                        \n                    >\n                        Ελληνικά\n                    </option>\n                    \n                    <option\n                        value=\"en\"\n                        \n                        selected=\"selected\"\n                        \n                    >\n                        English\n                    </option>\n                    \n                    <option\n                        value=\"es\"\n                        \n                    >\n                        Español\n                    </option>\n                    \n                    <option\n                        value=\"es_419\"\n                        \n                    >\n                        Español (Latinoamérica)\n                    </option>\n                    \n                    <option\n                        value=\"fi\"\n                        \n                    >\n                        Suomi\n                    </option>\n                    \n                    <option\n                        value=\"fr\"\n                        \n                    >\n                        Français\n                    </option>\n                    \n                    <option\n                        value=\"he\"\n                        \n                    >\n                        עברית\n                    </option>\n                    \n                    <option\n                        value=\"hr\"\n                        \n                    >\n                        Hrvatski\n                    </option>\n                    \n                    <option\n                        value=\"hu\"\n                        \n                    >\n                        Magyar\n                    </option>\n                    \n                    <option\n                        value=\"it\"\n                        \n                    >\n                        Italiano\n                    </option>\n                    \n                    <option\n                        value=\"ja\"\n                        \n                    >\n                        日本語\n                    </option>\n                    \n                    <option\n                        value=\"nb\"\n                        \n                    >\n                        Norsk\n                    </option>\n                    \n                    <option\n                        value=\"nl\"\n                        \n                    >\n                        Nederlands\n                    </option>\n                    \n                    <option\n                        value=\"pl\"\n                        \n                    >\n                        Polski\n                    </option>\n                    \n                    <option\n                        value=\"pt_BR\"\n                        \n                    >\n                        Português (Brasil)\n                    </option>\n                    \n                    <option\n                        value=\"pt_PT\"\n                        \n                    >\n                        Português (Portugal)\n                    </option>\n                    \n                    <option\n                        value=\"ro\"\n                        \n                    >\n                        Română\n                    </option>\n                    \n                    <option\n                        value=\"ru\"\n                        \n                    >\n                        Русский\n                    </option>\n                    \n                    <option\n                        value=\"sk\"\n                        \n                    >\n                        Slovenčina\n                    </option>\n                    \n                    <option\n                        value=\"sr\"\n                        \n                    >\n                        Srpski\n                    </option>\n                    \n                    <option\n                        value=\"tr\"\n                        \n                    >\n                        Türkçe\n                    </option>\n                    \n                    <option\n                        value=\"uk\"\n                        \n                    >\n                        Українська\n                    </option>\n                    \n                    <option\n                        value=\"zh_CN\"\n                        \n                    >\n                        简体中文\n                    </option>\n                    \n                    <option\n                        value=\"zh_TW\"\n                        \n                    >\n                        正體中文\n                    </option>\n                    \n                </select>\n\n                <!-- Pass our current url path as next. -->\n                <input type=\"hidden\" name=\"next\" value=\"/login\">\n\n                <!-- Pass query_params over to /language via POST. -->\n                <input type=\"hidden\" name=\"q\" value=\"next=%2F\">\n\n                <input type=\"submit\" value=\"Go\">\n            </div>\n        </fieldset>\n    </form>\n</div>\n    <div id=\"archdev-navbar\">\n    <ul>\n        \n            \n            <li><a href=\"/\">AUR Home</a></li>\n        \n\n        \n        <li><a href=\"/packages\">Packages</a></li>\n\n        \n            \n            <li><a href=\"/register\">Register</a></li>\n\n            \n            <li><a href=\"/login?next=/\">Login</a></li>\n        \n    </ul>\n</div>\n\n    \n\n<div id=\"dev-login\" class=\"box\">\n    <h2>AUR Login</h2>\n\n    \n        \n            <form method=\"post\" action=\"/login?next=/\">\n                <fieldset>\n                    <legend>Enter login credentials</legend>\n\n                    \n                        <ul class=\"errorlist\">\n                            \n                                <li>Bad username or password.</li>\n                            \n                        </ul>\n                    \n\n                    <p>\n                        <label for=\"id_username\">\n                            User name or primary email address:\n                        </label>\n\n                        <input id=\"id_username\" type=\"text\" name=\"user\" size=\"30\"\n                               maxlength=\"254\" autofocus=\"autofocus\"\n                               required=\"required\" value=\"LJNielsenDk\">\n                    </p>\n\n                    <p>\n                        <label for=\"id_password\">\n                            Password:\n                        </label>\n                        <input id=\"id_password\" type=\"password\" name=\"passwd\"\n                               size=\"30\" required=\"required\">\n                    </p>\n\n                    <p>\n                        <input id=\"id_remember_me\" type=\"checkbox\" name=\"remember_me\"\n                            \n                                checked=\"checked\"\n                            \n                        >\n                        <label for=\"id_remember_me\">\n                            Remember me\n                        </label>\n                    </p>\n\n                    <p>\n                        <input class=\"button\" type=\"submit\"\n                               value=\"Login\">\n                        <a href=\"/passreset\">\n                            [Forgot Password]\n                        </a>\n                        <input id=\"id_referer\" type=\"hidden\" name=\"referer\"\n                               value=\"https://aur.archlinux.org\">\n                        <input type=\"hidden\" name=\"next\" value=\"/\">\n                    </p>\n\n                </fieldset>\n            </form>\n        \n    \n</div>\n\n\n\n    <div id=\"footer\">\n    <p>\n        aurweb <a href=\"https://gitlab.archlinux.org/archlinux/aurweb/-/commits/v6.1.9\">v6.1.9</a>\n        \n    </p>\n    <p>Report issues <a href=\"https://gitlab.archlinux.org/archlinux/aurweb/-/issues/new?issuable_template=Bug\">here</a>.</p>\n    <p>Copyright &copy; 2004-2023 aurweb Development Team.</p>\n    <p>AUR packages are user produced content. Any use of the provided files is at your own risk.</p>\n</div>\n</div>"
```

**Comment by aarnt** [comment]
Hi. Aur vote is working for me. Do you use dannish chars in your password? It may be something related to it.

**Comment by LJNielsenDk** [comment]
Huh. Apparently replying to the notification email didn't work for commenting here like the mail said it should have. Anyway:

> Hi. Aur vote is working for me. Do you use dannish chars in your password? It may be something related to it.

No, but there are some ASCII special characters. ~~I'll try to change them and see if it helps any.~~
It was `+` that it didn't like.
I also tested some of the others, but I forgot to make a list of which ones I did test. At the least I tested these `~_-@$%^'.[{?` as working fine. If for some strange reason positioning in the password matters I replaced all the specials in the original password with _ and then tested each of the ones I'd used one at a time as the second character of the password just in case there were special rules for the first character.
